vault-plugins/shared/kafka/broker.go

```diff
@@ -229,61 +235,60 @@ func (mb *MessageBroker) GetKafkaTransactionalProducer() *kafka.Producer {
 	return mb.getTransactionalProducer()
 }
 
-func (mb *MessageBroker) GetConsumer(consumerGroupID, topicName string, autocommit bool) *kafka.Consumer {
+func (mb *MessageBroker) GetUnsubscribedRunConsumer(consumerGroupID string) *kafka.Consumer {
 	brokers := strings.Join(mb.config.Endpoints, ",")
 	c, err := kafka.NewConsumer(&kafka.ConfigMap{
 		"bootstrap.servers":        brokers,
 		"group.id":                 consumerGroupID,
 		"auto.offset.reset":        "earliest",
-		"enable.auto.commit":       autocommit,
+		"enable.auto.commit":       false,
 		"isolation.level":          "read_committed",
 		"go.events.channel.enable": true,
 	})
 	if err != nil {
 		panic(err)
 	}
-	err = c.SubscribeTopics([]string{topicName}, nil)
+	return c
+}
+
+func (mb *MessageBroker) GetSubscribedRunConsumer(consumerGroupID, topicName string) *kafka.Consumer {
+	c := mb.GetUnsubscribedRunConsumer(consumerGroupID)
+	err := c.Subscribe(topicName, nil)
 	if err != nil {
 		panic(err)
 	}
-
 	return c
 }
 
-func (mb *MessageBroker) GetRestorationReader(topic string) *kafka.Consumer {
+// GetRestorationReader returns Unsubscribed for any topic consumer
+func (mb *MessageBroker) GetRestorationReader() *kafka.Consumer {
 	brokers := strings.Join(mb.config.Endpoints, ",")
 	c, err := kafka.NewConsumer(&kafka.ConfigMap{
 		"bootstrap.servers":        brokers,
 		"auto.offset.reset":        "earliest",
-		"group.id":                 false,
+		"group.id":                 "restoration_reader",
 		"enable.auto.commit":       false,
 		"isolation.level":          "read_committed",
 		"go.events.channel.enable": true,
 	})
 	if err != nil {
 		panic(err)
 	}
-	err = c.SubscribeTopics([]string{topic}, nil)
-	if err != nil {
-		panic(err)
-	}
 
 	return c
 }
 
 func (mb *MessageBroker) SendMessages(msgs []Message, sourceInput *SourceInputMessage) error {
-	if sourceInput != nil && sourceInput.IgnoreBody {
+	mb.logger.Debug("SendMessages - started")
+	defer mb.logger.Debug("SendMessages - exit")
+	if sourceInput != nil && (sourceInput.IgnoreBody || len(msgs) == 0) {
 		return mb.sendOffset(sourceInput)
 	}
 
 	if len(msgs) == 0 {
 		return nil
 	}
 
-	if len(msgs) == 1 && sourceInput == nil { // only single message without source
```

Comment:
Why we delete this?

Reply:
We have explored with Yury - using at the same time transactional commit reading and none-transactional leads to ambiguity in message offsets. So, we decide to use only transactional commit read.